cpp/dolfinx/fem/assemble_matrix_impl.h

```diff
@@ -534,51 +541,49 @@ void assemble_matrix(
         fn, coeffs, cstride, constants, cell_info0, cell_info1);
   }
 
+  std::function<std::uint8_t(std::size_t)> get_perm;
+  if (a.needs_facet_permutations())
+  {
+    mesh->topology_mutable()->create_entity_permutations();
+    const std::vector<std::uint8_t>& perms
+        = mesh->topology()->get_facet_permutations();
+    get_perm = [&perms](std::size_t i) { return perms[i]; };
+  }
+  else
+    get_perm = [](std::size_t) { return 0; };
+
+  mesh::CellType cell_type = mesh->topology()->cell_type();
+  int num_cell_facets
+      = mesh::cell_num_entities(cell_type, mesh->topology()->dim() - 1);
   for (int i : a.integral_ids(IntegralType::exterior_facet))
   {
     auto fn = a.kernel(IntegralType::exterior_facet, i);
     assert(fn);
     auto& [coeffs, cstride]
         = coefficients.at({IntegralType::exterior_facet, i});
     impl::assemble_exterior_facets(
-        mat_set, x_dofmap, x, a.domain(IntegralType::exterior_facet, i),
+        mat_set, x_dofmap, x, num_cell_facets,
+        a.domain(IntegralType::exterior_facet, i),
         {dofs0, bs0, a.domain(IntegralType::exterior_facet, i, *mesh0)}, P0,
         {dofs1, bs1, a.domain(IntegralType::exterior_facet, i, *mesh1)}, P1T,
-        bc0, bc1, fn, coeffs, cstride, constants, cell_info0, cell_info1);
+        bc0, bc1, fn, coeffs, cstride, constants, cell_info0, cell_info1,
+        get_perm);
   }
 
-  if (a.num_integrals(IntegralType::interior_facet) > 0)
+  for (int i : a.integral_ids(IntegralType::interior_facet))
```

Comment:
Happy to remove the if statement (as it is not good on multiple processes, ref: https://github.com/FEniCS/dolfinx/pull/3215).